R/readWDI.R

```diff
@@ -70,25 +75,54 @@ convertWDI <- function(x, subtype) {
   if (subtype %in% c("SP.POP.TOTL", "SP.POP.1564.TO", "NY.GDP.MKTP.PP.KD", "NV.AGR.TOTL.KD")) {
     # Change scale of indicators
     x <- x / 1e+6
+
     # Add Kosovo to Serbia
-    x["RS", , ] <- dimSums(x[c("RS", "XK"), , ], dim = 1, na.rm = TRUE)
-  } else {
-    vcat("Warning: Kosovo left out of conversion and has differing population values from FAO", verbosity = 2)
+    # !TODO could be improved to adjust to the fact that Kosovo data might be missing for some years
+    x["SRB", , ] <- dimSums(x[c("SRB", "XKK"), , ], dim = 1, na.rm = TRUE)
+    x <- x["XKK", , , invert = TRUE]
   }
 
-  getCells(x) <- countrycode::countrycode(getCells(x),
-                                          "iso2c",
-                                          "iso3c",
-                                          custom_match = c("JG" = "JEY"),
-                                          warn = FALSE)
+  # Normalization of GDP due to update units by the World Bank
+  if (subtype == "NY.GDP.MKTP.PP.KD") {
+    # All three lead to the same outputs
+    # wdi_l[is.na(wdi_n[,,1])]
+    # !TODO ask Johannes if it is normal
```

Comment:
This could be "ok", if the NAs are not for certain years only, but across all years . The "linear" option only works, if there is data for inter- and/or extrapolation. Not sure if that is the case here, would have to look at the specific NAs.